src/client.rs

```diff
@@ -469,7 +469,25 @@ where
         target: &'a TargetPath,
     ) -> Result<Box<dyn AsyncRead + Send + Unpin>> {
         let target_description = self.fetch_target_description(target).await?;
-        self.remote.fetch_target(target, &target_description).await
+
+        if self.tuf.root().consistent_snapshot() {
+            let (_, value) = crypto::hash_preference(target_description.hashes())?;
+
+            let mut components = target.components();
+            let file_name = match components.pop() {
+                Some(file_name) => file_name,
+                None => {
+                    return Err(Error::IllegalArgument("Path cannot be empty".into()));
+                }
+            };
+            components.push(format!("{}.{}", value, file_name));
+
+            let target = TargetPath::new(components.join("/"))?;
```

Comment:
It doesn't really feel like this is the right place to be mangling the TargetPath. It seems like this translation is supposed to happen when you translate a VirtualTargetPath into a TargetPath? Does that not work for some reason?